src/panels/config/automation/trigger/types/ha-automation-trigger-webhook.ts

```diff
@@ -11,14 +21,63 @@ export class HaWebhookTrigger extends LitElement {
 
   @property() public trigger!: WebhookTrigger;
 
+  @state() private _config?: AutomationConfig;
+
+  private _unsub?: UnsubscribeFunc;
+
   public static get defaultConfig() {
     return {
-      webhook_id: "",
+      webhook_id: undefined,
     };
   }
 
+  connectedCallback() {
+    super.connectedCallback();
+    const details = { callback: (config) => this._automationUpdated(config) };
+    fireEvent(this, "subscribe-automation-config", details);
+    this._unsub = (details as any).unsub;
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    if (this._unsub) {
+      this._unsub();
+    }
+  }
+
+  private _automationUpdated(config?: AutomationConfig) {
+    this._config = config;
+  }
+
+  private _generateWebhookId(): string {
+    // The webhook_id should be treated like a password. Generate a default
+    // value that would be hard for someone to guess. This generates a
+    // 144-bit random value. The output is a 24 character url-safe string.
+    const randomBytes = crypto.getRandomValues(new Uint8Array(18));
+    const base64Str = btoa(String.fromCharCode(...randomBytes));
+    const urlSafeId = base64Str.replace(/\+/g, "-").replace(/\//g, "_");
+
+    // Include the automation name to give the user context about what the
+    // webhook_id is used for.
+    const lowerAlias = this._config?.alias?.toLowerCase() || "";
+    const urlSafeAlias = lowerAlias
```

Comment:
We have a slugify helper that you can use.

Reply:
Thanks for the tip. Implemented in [6cc090b](https://github.com/home-assistant/frontend/pull/11568/commits/6cc090bc865b422a53216715ff059f7847092a99).